### ISSUE TITLE
Assert keyword

### ISSUE THREAD

**Issue by DelvalletQuentin** [beanshell/beanshell#27]
Hello,

I have been using beanshell for some school project and I'm kind of stuck with the keyword "assert". As I found, it is not implemented in beanshell and it is suggested to implement our own [in the wiki](https://github.com/beanshell/beanshell/wiki/Adding-BeanShell-Commands#getting-the-invocation-text).

So I did that and it works well. Until I found out someone in my team is using the keyword that way : 
`assert i != 0;`

Instead of : 
`assert (i != 0);`

And with the first example it doesn't work since my own assert implementation act as a method. Is there any way we can have a "true" assert keyword in beanshell? And is there a temporary fix I can do to make the first example work?

My assert implementation in beanshell looks like this for now : 
```
assert(boolean condition) { 
    if (!condition)
        throw new AssertionError();
}
```

**Comment by stain** [comment]
Yes, defining your own `assert` command would basically just be making a function that takes a single argument - it would not be greedy like Java's own `assert` keyword.

Note that using `assert` in Java is not best practice for unit testing, as the compiler often disables assertions; should we in Beanshell still encourage the use of `assert` or rather make it easy to use say JUnit from Beanshell?

**Comment by DelvalletQuentin** [comment]
Well, what we actually do is "an Automatic bug diagnosis" Java program. We generate a cause-effect chain (see [Andreas Zeller publications](http://swtv.kaist.ac.kr/courses/cs750b-sw-model-checking-fall-06/cs750b-sw-model-checking-fall-06/p201-zeller.pdf)) that lead to a bug.

Since we need to develop the system, we have to somehow tell our program that there is actually a bug in the code (since bugs don't automatically lead to exception and/or crashes). So we use Java asserts to test (not test as in unit tests) our program and our generated cause-effect chain.

Moreover, we don't have full control in the code we give to our system since we take other students' code as proof that our program works well, so I guess if someone wants to use Java's assert or JUnit methods, it's up to him and we only have to handle the most cases we can.

Anyway, of course it would be nice for Beanshell to handle the most syntaxes and keywords it could, but I understand the current situation, since the assert keyword is uncommon.

Also, we bypassed the initial problem. Since we already had dynamic code modifications thanks to [Javaparser](https://github.com/javaparser/javaparser) in our program, we added a pre-processor for assert statements that simply adds parenthesis.
